streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -493,22 +493,18 @@ private void handleStreamsUncaughtException(final Throwable throwable,
                 closeToError();
                 break;
             case SHUTDOWN_APPLICATION:
+                if (countStreamThread(StreamThread::isRunning) == 1) {
+                    log.warn("Adding thread to communicate the shutdown. No processing will be done on this thread");
```

Comment:
If we add a thread before shutting down the application the thread with the exception can shutdown